File: gcj_2015/QR_A.py
#!/usr/bin/env python

# Google Code Jam 2015 Qualification Round
# Problem A. Standing Ovation

import sys

def solve(m, s):
  ans = 0
  tot = 0
  for level in range(0, m+1):
    n = int(s[level])
    if n > 0:
      need = max(0, level - tot)
      ans += need
      tot += need + n
  return ans

n = int(sys.stdin.readline())
for t in range(1, n+1):
  args = []
  line = sys.stdin.readline().rstrip("\n").split(' ')
  ans = solve(int(line[0]), line[1])
  print("Case #" + str(t) + ": " + str(ans))

File: gcj_2015/QR_A.clj
(use '[clojure.string :only (join split)])

(defn solve[t]
  (def input (split (read-line) #"\s"))
  (def n (read-string (first input)))
  (def s (map read-string (split (second input) #"")))
  (defn add[level n sum s]
    (def p (max 0 (- level n)))
    (if (empty? s)
      sum
      (add (+ level 1) (+ n (first s) p) (+ sum p) (rest s))))
  (add 0 0 0 s))

(defn output-result[t ans] (println (format "Case #%d: %d" t ans)))
(def T (read-string (read-line)))
(doall (map (fn[t] (output-result (+ t 1) (solve t))) (range 0 T)))
